### PR TITLE
Separate interactive data providers

### LINKED ISSUE #108: Add DataProvider.interactive() method
It would be good if a data provider can tell whether it interacts with a user or not. Let's add a new `DataProvider.interactive()` method which returns true if a provider talks to a user, and false otherwise. Some of the existing data providers, which can talk to a user, will need to be split to two.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Added a new `DataProvider.interactive()` method.
- Extracted interactive part of the existing data providers to new `AskAboutUnpatchedVulnerabilities` and `AskAboutSecurityTeam` data providers.
- Added tests.
- Updated the demo tool.

This closes #108

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        break;
+      case NO:
+        values.update(new BooleanValue(HAS_SECURITY_TEAM, false));
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format("Hey! You gave me an unexpected answer: %s", answer));
+    }
+
+    // TODO: store the answer in the SecurityTeamStorage
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:33]: Is there an issue tracking this?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * This data provider asks a user about security teams.
+ */
+public class AskAboutSecurityTeam<T> extends AbstractInteractiveDataProvider<T> {
+
+  @Override
+  protected AskAboutSecurityTeam<T> ask(T object, ValueSet values) {
+    String question = String.format(
+        "Does project '%s' have a security team? Say yes, no, or skip, please.", object);
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:18]: How is skip handled here? I mean in the switch case?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutUnpatchedVulnerabilities.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        unpatchedVulnerabilities.add(vulnerability);
+        answer = new YesNoQuestion(callback, "One more?").ask();
+      } while (answer == YES);
+    }
+
+    Value<Vulnerabilities> vulnerabilities = knownVulnerabilities(values);
+    vulnerabilities.get().add(unpatchedVulnerabilities);
+    values.update(vulnerabilities);
+
+    // TODO: store the entered vulnerabilities somewhere
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutUnpatchedVulnerabilities.java:49]: Is there an issue tracking this?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        values.update(new BooleanValue(HAS_SECURITY_TEAM, false));
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format("Hey! You gave me an unexpected answer: %s", answer));
+    }
+
+    // TODO: store the answer in the SecurityTeamStorage
+
+    return null;
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:35]: `return this;` ?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java (hunk 1 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:33]: No, we don't have an issue for that. To be honest, I am not sure if it should be implemented. The interactive providers are not a priority for now.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:18]: Good point. I'll add a case for `SKIP`.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutUnpatchedVulnerabilities.java (hunk 3 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutUnpatchedVulnerabilities.java:49]: No, we don't have an issue for that. To be honest, I am not sure if it should be implemented. The interactive providers are not a priority for now.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java (hunk 4 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/interactive/AskAboutSecurityTeam.java:35]: Yup!

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala ! I've addressed your comments.

**Pull request merged**